model-engine/model_engine_server/inference/batch_inference/vllm_batch.py

```diff
@@ -414,6 +420,30 @@ async def generate_with_vllm(
         if return_token_log_probs:
             output.tokens = tokens
 
+        tags = [
+            f"model:{model}",
+            f"use_tool:{use_tool}",
+        ]
+        statsd.increment(
+            "model_engine.batch_inference.vllm.generation_count",
+            tags=tags,
+        )
+        statsd.increment(
+            "model_engine.batch_inference.vllm.token_count.total",
+            num_prompt_tokens + num_completion_tokens,
```

Comment:
Is this possible to just compute when querying metrics later?

Reply:
I think if we do increment we can just add them later, but if we want to do something more fancy like histogram/distribution then we need to output the total separately

Reply:
yes it is possible, also things like Sean said